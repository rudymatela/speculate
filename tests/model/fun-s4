max expr size  =    4
  |- on ineqs  =    3
  |- on conds  =    3
max  #-tests   = 1000
min  #-tests   =   50  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Int  (holes: Int)
_ :: [Int]  (holes: [Int])
_ :: Int -> Int  (holes: Int -> Int)
map :: (Int -> Int) -> [Int] -> [Int]
id :: Int -> Int
(.) :: (Int -> Int) -> (Int -> Int) -> Int -> Int

     id x == x
  f (g x) == (f . g) x
map id xs == xs
   f . id == f
   id . f == f


