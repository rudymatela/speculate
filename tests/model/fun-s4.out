max expr size  =    4
  |- on ineqs  =    3
  |- on conds  =    3
max  #-tests   = 1000
min  #-tests   =   50  (to consider p ==> q true)
max  #-vars    =    2  (for inequational and conditional laws)

_ :: Int
_ :: [Int]
_ :: Int -> Int
map :: (Int -> Int) -> [Int] -> [Int]
id :: Int -> Int
(.) :: (Int -> Int) -> (Int -> Int) -> Int -> Int

     id x == x
  f (g x) == (f . g) x
map id xs == xs
   f . id == f
   id . f == f


